### ISSUE TITLE
ghc-toolkit> ghc: mmap 131072 bytes at 0x0: Operation not permitted

### ISSUE THREAD

**Issue by BebeSparkelSparkel** [tweag/asterius#425]
```bash
> stack build
ghc-toolkit> configure (lib)
ghc-toolkit> Configuring ghc-toolkit-0.0.1...
ghc-toolkit> ghc: mmap 131072 bytes at 0x0: Operation not permitted
ghc-toolkit> ghc: Try specifying an address with +RTS -xm<addr> -RTS
ghc-toolkit> ghc: internal error: m32_allocator_init: Failed to map
ghc-toolkit>     (GHC version 8.6.5 for x86_64_apple_darwin)
ghc-toolkit>     Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
ghc-toolkit> readCreateProcess: /Users/williamrusnack/.stack/programs/x86_64-osx/ghc-custom-asterius-8.6.5/bin/runghc "--ghc-arg=-Ighc-libdir/include" "--ghc-arg=-Iinclude-private" "genapply/Main.hs" (exit -6): failed

--  While building package ghc-toolkit-0.0.1 using:
      /Users/williamrusnack/asterius/ghc-toolkit/.stack-work/dist/x86_64-osx-custom-asterius/Cabal-2.4.0.1/setup/setup --builddir=.stack-work/dist/x86_64-osx-custom-asterius/Cabal-2.4.0.1 configure --user --package-db=clear --package-db=global --package-db=/Users/williamrusnack/.stack/snapshots/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/pkgdb --package-db=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/pkgdb --libdir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/lib --bindir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/bin --datadir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/share --libexecdir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/libexec --sysconfdir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/etc --docdir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/doc/ghc-toolkit-0.0.1 --htmldir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/doc/ghc-toolkit-0.0.1 --haddockdir=/Users/williamrusnack/asterius/.stack-work/install/x86_64-osx-custom-asterius/1071f5c15375e08399b973bf1b6077582ecb3364ed3e70e0f9e2bd6808422644/8.6.5/doc/ghc-toolkit-0.0.1 --dependency=Cabal=Cabal-2.4.1.0-3JC906oWQ73Ah7bYkIcI4J --dependency=base=base-4.12.0.0 --dependency=containers=containers-0.6.0.1 --dependency=directory=directory-1.3.3.0 --dependency=filepath=filepath-1.4.2.1 --dependency=ghc=ghc-8.6.5 --dependency=process=process-1.6.5.0 --exact-configuration --ghc-option=-fhide-source-paths
    Process exited with code: ExitFailure 1
Progress 1/2
```

OS: macOS Catalina 10.15.2 (19C57)

```bash
> stack --version
Version 2.1.3, Git revision 0fa51b9925decd937e4a993ad90cb686f88fa282 (7739 commits) x86_64 hpack-0.31.2
```

stack resolver: lts-14.20

**Comment by hsyl20** [comment]
This bug is tracked upstream: https://gitlab.haskell.org/ghc/ghc/issues/17353#note_235457

It is fixed in GHC 8.8 but sadly not in 8.6.5

**Comment by BebeSparkelSparkel** [comment]
Thank you. I will try nightly
